flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyGroupRangeOffsets.java

```diff
@@ -141,8 +150,7 @@ public KeyGroupRange getKeyGroupRange() {
     private int computeKeyGroupIndex(int keyGroup) {
         int idx = keyGroup - keyGroupRange.getStartKeyGroup();
         if (idx < 0 || idx >= offsets.length) {
-            throw new IllegalArgumentException(
-                    "Key group " + keyGroup + " is not in " + keyGroupRange + ".");
+            throw newIllegalKeyGroupException(keyGroup, keyGroupRange);
```

Comment:
I think we are double throwing something here. Maybe let `newIllegalKeyGroupException` method only return the `IllegalStateException` instead of throwing it.

Reply:
🤦 That was the original intention